src/overlays/actors/ovl_En_Gs/z_en_gs.c

```diff
@@ -74,78 +109,1007 @@ static DamageTable D_8099A3D8 = {
+            this->unk_216 = 0;
+        }
+
+        if (this->actor.playerHeightRel < -12000.0f) {
+            Actor_Spawn(&globalCtx->actorCtx, globalCtx, ACTOR_EN_BOM, this->actor.world.pos.x, this->actor.world.pos.y,
+                        this->actor.world.pos.z, 0, this->actor.world.rot.y, 0, 0);
+            Actor_MarkForDeath(&this->actor);
+            sp7C = 0;
+        }
+    }
+
+    return sp7C;
+}
+
+void func_80999A8C(EnGs* this, GlobalContext* globalCtx) {
+    if (this->unk_1D4-- <= 0) {
+        Actor_MarkForDeath(&this->actor);
+    }
+}
 
-extern ColliderCylinderInit D_8099A3A0;
-extern CollisionCheckInfoInit2 D_8099A3CC;
-extern DamageTable D_8099A3D8;
-extern InitChainEntry D_8099A404[];
+void func_80999AC0(EnGs* this) {
+    static Vec3s D_8099A450 = { 0, 0, 0 };
+    static Vec3f D_8099A458 = { 1.0f, 1.0f, 1.0f };
+    s32 i;
 
-extern UNK_TYPE D_06000950;
+    for (i = 0; i < ARRAY_COUNT(this->unk_19E); i++) {
+        this->unk_19E[i] = D_8099A450;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/func_80997A90.s")
+    for (i = 0; i < ARRAY_COUNT(this->unk_1B0); i++) {
+        this->unk_1B0[i] = D_8099A458;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/func_80997AFC.s")
+void func_80999B34(EnGs* this) {
+    if (this->unk_216 > 0) {
+        if (this->unk_218 == 0) {
+            func_8016566C(this->unk_216);
+            this->unk_218 = this->unk_216;
+        } else {
+            func_80165658(this->unk_218);
+        }
+    } else if (this->unk_218 > 0) {
+        Math_StepToS(&this->unk_218, this->unk_216, 20);
+        func_80165658(this->unk_218);
+        if (this->unk_218 <= 0) {
+            func_80165690();
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/EnGs_Init.s")
+void func_80999BC8(Actor* thisx, GlobalContext* globalCtx2) {
+    GlobalContext* globalCtx = globalCtx2;
+    EnGs* this = THIS;
+    s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/EnGs_Destroy.s")
+    if (this->actor.isTargeted && !func_801A5100()) {
+        this->unk_19D = 0;
+        this->unk_19A |= 1;
+        func_80999AC0(this);
+        this->unk_19C = 0;
+        func_80998704(this, globalCtx);
+        this->unk_208 = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/func_80997D14.s")
+    if (this->actor.params == ENGS_1) {
+        Actor_SetHeight(&this->actor, 34.5f);
+    } else {
+        Actor_SetHeight(&this->actor, 23.0f);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/func_80997D38.s")
+    if (this->unk_21A > 0) {
+        this->unk_21A--;
+        if (this->unk_21A <= 0) {
+            this->unk_21A = 0;
+        }
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Gs/func_80997DEC.s")
+    if (!(this->unk_19A & (0x200 | 0x10)) && (this->unk_21A == 0)) {
+        if (this->collider.base.acFlags & 2) {
```

Comment:
These flags should use defines from `z64collision_check.h`: this one is `AC_HIT`.